### PR TITLE
use uv in tox

### PR DESCRIPTION
**Pull request by alangenfeld**
Use uv for env setup in tox to improve performance. 

## How I Tested These Changes

bk

### CODE REVIEW

**Comment by alangenfeld** [comment]
Current dependencies on/for this PR:
* `master`
  * **PR #19871** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/19871?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/19871?utm_source=stack-comment).

**Comment by alangenfeld** [comment]
for uv 0.1.2, currently hitting what I believe some sort of race condition tracked https://github.com/astral-sh/uv/issues/1508

File: python_modules/dagster/dagster/_core/test_utils.py [hunk 1]
```diff
@@ -700,3 +702,11 @@ def raise_exception_on_warnings():
...
         # pendulum sometimes raises DeprecationWarning on python3.12
         warnings.filterwarnings("ignore", category=DeprecationWarning, module="pendulum")
+
+
+def ensure_dagster_tests_import() -> None:
+    dagster_package_root = (Path(dagster_init_py) / ".." / "..").resolve()
+    assert (
+        dagster_package_root / "dagster_tests"
+    ).exists(), "Could not find dagster_tests where expected"
+    sys.path.append(dagster_package_root.as_posix())
```
Comment (alangenfeld) [inline python_modules/dagster/dagster/_core/test_utils.py:712]: X

File: python_modules/dagster/tox.ini [hunk 2]
```diff
@@ -8,7 +8,11 @@ setenv =
   windows: COVERAGE_ARGS =
   STRICT_GRPC_SERVER_PROCESS_WAIT = "1"
 passenv = CI_* COVERALLS_REPO_TOKEN AWS_SECRET_ACCESS_KEY AWS_ACCESS_KEY_ID BUILDKITE* DAGSTER_DOCKER_* GRPC_SERVER_HOST
-install_command = python -m pip install {opts} {packages} --config-settings editable_mode=compat
+install_command =
+  ; pyright/mypy invocations type_signature_tests in need commpat mode installs to find types correctly
+  type_signature_tests: python -m pip install {opts} {packages} --config-settings editable_mode=compat
+  !type_signature_tests: uv pip install {opts} {packages}
+
```
Comment (alangenfeld) [inline python_modules/dagster/tox.ini:15]: X

**Comment by sbquinlan** [comment]
lgtm. "Python Sean" (@smackesey) to stamp

**Review by smackesey — APPROVED** [review]
Let's merge it and try to fix the special cases over the next week or so. Pretty sure the devs will help us work through the issues.

**Pull request merged**